### PR TITLE
Added possibility to batch remove a tag

### PR DESCRIPTION
**Pull request by beni71**
Pull Request for Issue #23185 .

### Summary of Changes
In the articles view there is a batch action which can add/assign one tag from the tag list to the selected articles. This PR introduces the possibility to remove/unassign one tag from the selected articles. It also extends batch processing for categories, contacts and news feeds.

How it looks like:

articles-batch-tag-removing

### Testing Instructions
1. Make sure your Joomla tag system is active and the plugin "Behaviour - Taggable" is enabled.
2. Create a new article and in the detail view add/assign two tags, lets say "Tag1" and "Tag2".
3. Save and create a second article, add/assign it to tag "Tag2" and "Tag3".
4. Save and close.
5. In the articles list select both articles, click batch action and remove "Tag2". Apply batch processing.
6. Open the first article details and check whether the "Tag2" is removed. The tag "Tag1" must be still assigned.
7. Open the second article details and check whether the "Tag2" is removed. The tag "Tag3" must be still assigned.
8. A similar test should be done for categories, contacts and news feeds.

### Actual result BEFORE applying this Pull Request
It is not possible to remove/unassign a tag from an article via batch processing.

### Expected result AFTER applying this Pull Request
It is possible to remove/unassign a tag from an article via batch processing.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

There are new text keys that need to be translated.

### CODE REVIEW

File: administrator/language/en-GB/lib_joomla.ini [hunk 1]
```diff
@@ -376,8 +376,11 @@ JLIB_HTML_BATCH_MOVE="Move"
 JLIB_HTML_BATCH_MOVE_QUESTION="Action to Perform"
 JLIB_HTML_BATCH_NO_CATEGORY="- Don't copy or move -"
 JLIB_HTML_BATCH_NOCHANGE="- Keep original Access Levels -"
-JLIB_HTML_BATCH_TAG_LABEL="Add Tag"
-JLIB_HTML_BATCH_TAG_NOCHANGE="- Keep original Tags -"
+JLIB_HTML_BATCH_TAG_MENU_LABEL="Add or remove a Tag"
+JLIB_HTML_BATCH_TAG_NO_CHANGE="- Don't add or remove Tags -"
+JLIB_HTML_BATCH_TAG_ADDREMOVE_QUESTION="Action to Perform"
+JLIB_HTML_BATCH_TAG_ADD="Add"
+JLIB_HTML_BATCH_TAG_REMOVE="Remove"
```
Comment (richard67) [inline administrator/language/en-GB/lib_joomla.ini]:
```suggestion
JLIB_HTML_BATCH_TAG_ADD="Add"
JLIB_HTML_BATCH_TAG_ADDREMOVE_QUESTION="Action to Perform"
JLIB_HTML_BATCH_TAG_MENU_LABEL="Add or remove a Tag"
JLIB_HTML_BATCH_TAG_NO_CHANGE="- Don't add or remove Tags -"
JLIB_HTML_BATCH_TAG_REMOVE="Remove"
```
Alpha ordering language strings.

**Comment by brianteeman** [comment]
This is a new feature isn't it?

File: build/media_source/layouts/js/joomla/html/batch/batch-tag-addremove.es6.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/**
+ * @copyright  (C) 2018 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline build/media_source/layouts/js/joomla/html/batch/batch-tag-addremove.es6.js]:
```suggestion
 * @copyright  (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
```

its a new file

**Comment by richard67** [comment]
> This is a new feature isn't it?

@brianteeman To me it seems so, and the referred issue also was labelled as feature and not as bug. Am just clarifying with other maintainers what they think.

**Comment by richard67** [comment]
@beni71 Thank you very much for this PR. Due to our development roadmap, new features will go into version 5.0. Before you get shocked now: 5.0 will be released at the same time as 4.4 in October this year.

4.4 will serve as a bridge version for updating to 5 in the same way as the 3.10 was for 4, so we don't include new features into 4.4 if they are not needed for a smooth update to 5.

4.3.0 had feature freeze when it was in beta phase, and with patch versions we do not include new features in general.

This all for your understanding so you are not too disappointed.

Could you rebase this PR to 5.0-dev? Let us know here if you need help or advise with that.

Or maybe it is better if you create a new branch based on the 5.0-dev branch and make the PR again with that new branch, so your 4.3-dev branch will not be rebased to 5.0-dev because that could be confusing in future.

Thanks in advance.

**Comment by beni71** [comment]
Ok, so I can close this PR.

**Pull request closed without merging**